sound/soc/codecs/hdac_hda.c

```diff
@@ -603,12 +607,7 @@ static int hdac_hda_dev_probe(struct hdac_device *hdev)
 
 static int hdac_hda_dev_remove(struct hdac_device *hdev)
 {
-	struct hdac_hda_priv *hda_pvt;
-
-	hda_pvt = dev_get_drvdata(&hdev->dev);
-	if (hda_pvt && hda_pvt->codec.registered)
-		cancel_delayed_work_sync(&hda_pvt->codec.jackpoll_work);
-
+	/* empty function needed for hda_codec_driver_remove() */
```

Comment:
I wasn't able to follow the commit message, the comment above and where the cancel delayed work would be done now?
@kv2019i maybe a couple of explanations would help here?

Reply:
Ack @plbossart . I'll do another round and add a bit more comments. This is somewhat hard to follow with the extended HDA bus and ASoC, laid on top of bare HDA interfaces.